### DISCUSSION TITLE
I can't get the meaning of ${Element.classFullName} in 'Select Top-Level Elements in a particular Model' examples

### DISCUSSION THREAD

**Discussion by Adu2120** [iTwin/itwinjs-core#4652] [Q&A] [answered]
In the below code, I am unable to get the meaning of '${Element.classFullName}'.

`const modelId: Id64String = IModelDb.repositoryModelId;
iModel.withPreparedStatement("SELECT ECInstanceId AS id FROM ${Element.classFullName} WHERE Model.Id=:modelId AND Parent.Id IS NULL", (statement: ECSqlStatement) => {
  statement.bindId("modelId", modelId);
  while (DbResult.BE_SQLITE_ROW === statement.step()) {
    // do something with each row
  }
});`

[Reference from website](https://www.itwinjs.org/learning/backend/ecsql-queries/#select-top-level-elements-in-a-particular-model)

**Comment by ColinKerr** [comment, marked as answer]
This is the name of the class you want to select.  For example if you wanted to select from all elements you would use `BisCore.Element` you can use the formula: `<schemaName>.<ClassName>`.

In the code snippet you are referencing gets the full class name from the TypeScript [Element](https://www.itwinjs.org/reference/core-backend/elements/element/) class, which inherits [classFullName from the base Entity class](https://www.itwinjs.org/reference/core-backend/schema/entity/#classfullname).  

When you call the [IModelDb.Elements.getElement](https://www.itwinjs.org/reference/core-backend/imodels/imodeldb/imodeldb.elements/getelement/) method it will return an object whose TypeScript class matches the class of the Element as stored in the iModel.  This element will return the correct full name for the class when you call `classFullName`.

To browse the standard classes available in BIS check out our [docs](https://www.itwinjs.org/bis/domains/) which include the definition for every standard class, e.g. [BisCore.Element](https://www.itwinjs.org/bis/domains/biscore.ecschema/#element)


 > NOTES: 
 >  - both `.` and `:` are valid separators between schema and class name in ECSql.
 >  - Schemas have both a name and an alias and either one can be used in ECSql. e.g. The schema `BisCore` has an alias of `bis`
 >  - Names are case insensitive in ECSql so `biscore.ELEMENT` is equivalent to `BisCore.Element`

**Reply by Adu2120** [reply]
Thanks!!
